python/cudf/cudf/core/frame.py

```diff
@@ -496,8 +496,8 @@ def get_column_values_na(col):
                     ),
                 ):
                     raise NotImplementedError(
-                        f"{dtype} is not yet supported via "
-                        "`__cuda_array_interface__`"
+                        f"{dtype} is not yet supported to be exported to"
+                        "a cupy array"
```

Comment:
There is a different version of similar error in `find_common_type`, the reason we cannot have it just there is someone can pass multiple identical type complex types to `find_common_type`, we don't want to error in those cases, or else `concat` code-flow would break. Here it's a guard to not even call `find_common_type` and is specific to `to_array`.

Reply:
> `f"{dtype} cannot be exposed as a cupy array"` ?

Done 👍